source/opt/eliminate_dead_members_pass.cpp

```diff
@@ -63,6 +76,10 @@ void EliminateDeadMembersPass::FindLiveMembers() {
         case SpvStorageClassOutput:
           MarkPointeeTypeAsFullUsed(inst.type_id());
           break;
+        case SpvStorageClassUniform:
+          if (structured_buffers.count(inst.result_id()) != 0)
```

Comment:
Instead of looking for the user type decoration, I would try to match the conditions used in spirv-cross.  There is a function in the instruction class that can help: https://github.com/KhronosGroup/SPIRV-Tools/blob/ba4b390c367e09edaad36f0376a87aa03620fd4b/source/opt/instruction.cpp#L367

You can write another function based on that function.  Then the check becomes:

```
if (inst.IsVulkanStorageBufferVariable()) {
  MarkPointeeTypeAsFullUsed(inst.type_id());
}
```

This check should be outside of the switch because it can be in the StorageBuffer storage class or the Uniform storage class.

This will make it more robust because it won't rely on a decoration that other spir-v generators may not use.  I think the same problem can come up if you started with a similar GLSL example.